backend/src/test/java/de/neuefische/backend/service/DoctorServiceTest.java

```diff
@@ -80,4 +107,230 @@ void addExistingDoctorTest() {
+    void addAppointmentToExistingDoctorTest() {
+        // GIVEN
+        DoctorDto doctorDto = DoctorDto.builder()
+                .firstName("Linda")
+                .lastName("Holder")
+                .specialty("Dentist")
+                .city("Bonn")
+                .appointmentDto(AppointmentDto.builder()
+                        .date("2021-11-08")
+                        .reasonForVisit("checkup")
+                        .build())
+                .build();
+
+        Doctor doctor = Doctor.builder()
+                .firstName("Linda")
+                .lastName("Holder")
+                .specialty("Dentist")
+                .city("Bonn")
+                .appointments(List.of(
+                        Appointment.builder()
+                                .date("2021-11-08")
+                                .reasonForVisit("checkup")
+                                .build()
+                ))
+                .build();
+
+        Doctor doctorInDb = Doctor.builder()
+                .firstName("Linda")
+                .lastName("Holder")
+                .specialty("Dentist")
+                .city("Bonn")
+                .appointments(List.of(
+                        Appointment.builder()
+                                .id("111111")
+                                .date("2020-10-02")
+                                .reasonForVisit("checkup")
+                                .build()
+                ))
+                .build();
+
+        Doctor updatedDoctor = Doctor.builder()
+                .firstName("Linda")
+                .lastName("Holder")
+                .specialty("Dentist")
+                .city("Bonn")
+                .appointments(List.of(
+                        Appointment.builder()
+                                .id("111111")
+                                .date("2020-10-02")
+                                .reasonForVisit("checkup")
+                                .build(),
+                        Appointment.builder()
+                                .id("123456")
+                                .date("2021-11-08")
+                                .reasonForVisit("checkup")
+                                .build()
+                ))
+                .build();
+
+        when(utilService.mapDoctorDtoToDoctor(doctorDto)).thenReturn(doctor);
+        when(utilService.mapAppointmentDtoToAppointment(doctorDto.getAppointmentDto())).thenReturn(doctor.getAppointments().get(0));
+        when(idService.generateId()).thenReturn("123456");
+        when(doctorRepo.findDoctorByFirstNameAndLastNameAndSpecialtyAndCity("Linda", "Holder", "Dentist", "Bonn")).thenReturn(doctorInDb);
+        when(doctorRepo.save(doctorInDb)).thenReturn(updatedDoctor);
+
+        // WHEN
+        Doctor actualDoctor = doctorService.addAppointment(doctorDto);
+
+        // THEN
+        assertEquals(updatedDoctor, actualDoctor);
+        verify(doctorRepo).save(doctorInDb);
```

Comment:
verify all mock calls
```suggestion
        verify(doctorRepo).save(doctorInDb);
```